distributed/executor.py

```diff
@@ -193,6 +193,22 @@ def release(self):
             self._cleared = True
             self.executor._dec_ref(tokey(self.key))
 
+    def __getstate__(self):
+        state = self.__dict__.copy()
+        state.update({'event': self.event.is_set(), 'status': self.status})
+        del state['executor']
+        return state
+
+    def __setstate__(self, state):
+        self.__dict__ = state
+        e = default_executor()
+        event = Event()
+        if state.pop('event'):
+            event.set()
+        e.futures[tokey(state['key'])] = {'event': event,
+                                          'status': state.pop('status')}
+        state['executor'] = e
```

Comment:
You need to send the `'client'` as `e.id` (a uuid string) rather than `e`.  If you run tests with the `-s` flag you'll get some extra debugging output, notably this error occurred:

```
distributed.core - ERROR - can't serialize <Executor: scheduler="127.0.0.1:42236">
Traceback (most recent call last):
  File "/home/mrocklin/workspace/distributed/distributed/core.py", line 274, in write
    frames = protocol.dumps(msg)
  File "/home/mrocklin/workspace/distributed/distributed/protocol.py", line 105, in dumps
    small_header, small_payload = dumps_msgpack(msg)
  File "/home/mrocklin/workspace/distributed/distributed/protocol.py", line 212, in dumps_msgpack
    payload = msgpack.dumps(msg, use_bin_type=True)
  File "/home/mrocklin/Software/anaconda/lib/python3.4/site-packages/pandas/msgpack/__init__.py", line 44, in packb
    return Packer(**kwargs).pack(o)
  File "pandas/msgpack/_packer.pyx", line 223, in pandas.msgpack._packer.Packer.pack (pandas/msgpack/_packer.cpp:3434)
  File "pandas/msgpack/_packer.pyx", line 225, in pandas.msgpack._packer.Packer.pack (pandas/msgpack/_packer.cpp:3276)
  File "pandas/msgpack/_packer.pyx", line 213, in pandas.msgpack._packer.Packer._pack (pandas/msgpack/_packer.cpp:3006)
  File "pandas/msgpack/_packer.pyx", line 184, in pandas.msgpack._packer.Packer._pack (pandas/msgpack/_packer.cpp:2433)
  File "pandas/msgpack/_packer.pyx", line 220, in pandas.msgpack._packer.Packer._pack (pandas/msgpack/_packer.cpp:3158)
TypeError: can't serialize <Executor: scheduler="127.0.0.1:42236">
```